runtime/src/leader_bank_status.rs

```diff
@@ -0,0 +1,167 @@
+use {
+    crate::bank::Bank,
+    solana_sdk::slot_history::Slot,
+    std::{
+        sync::{Arc, Condvar, Mutex, RwLock, Weak},
+        time::{Duration, Instant},
+    },
+};
+
+/// Tracks leader status of the validator node and notifies when:
+///     1. A leader slot begins
+///     2. A leader slot completes
+#[derive(Debug, Default)]
+pub struct LeaderBankStatus {
```

Comment:
naming: `LeaderBank{Notifier/Tracker/Provider}`?

Reply:
I like `Notifier` or `Tracker`. Going with `Notifier` since that's the main mechanism of interest

Reply:
Done: 1b378262622cd58ede1bf0f1695aa320043e0b55 + f1a927cd8aeb2b50658987c1b908d56f4b593e79